mzLib/MassSpectrometry/PeakIndexing/PeakSpline/XicSpline.cs

```diff
@@ -44,10 +45,10 @@ public abstract class XicSpline
         /// </summary>
         public void SetXicSplineXYData(ExtractedIonChromatogram xic, bool cycle = false, double start = -1, double end = -1)
         {
-            double[] peakRts = null;
+            float[] peakRts = null;
             if (cycle)
             {
-                peakRts = xic.Peaks.Select(p => (double)p.ZeroBasedScanIndex).ToArray();
+                peakRts = xic.Peaks.Select(p =>(float)p.ZeroBasedScanIndex).ToArray();
             }
```

Comment:
This was bugged to begin with. It should take in retention times, not scan indexes